ui/components/src/ui-components/inputs/ParsedInput.tsx

```diff
@@ -0,0 +1,112 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Inputs
+ */
+
+import * as React from "react";
+import classnames from "classnames";
+import { CommonProps, Input } from "@bentley/ui-core";
+import { ParseResults, SpecialKey } from "@bentley/ui-abstract";
+import "./ParsedInput.scss";
+
+/** Props for [[ParsedInput]] control
+ * @beta
+ */
+export interface ParsedInputProps extends CommonProps {
+  /** InitialValue which is used to restore input field if ESC is pressed */
+  initialValue: number;
+  /** Function used to format the value */
+  formatValue: (value: number) => string;
+  /** Function used to parse user input into a value */
+  parseString: (stringValue: string) => ParseResults;
+  /** Function to call when value is changed */
+  onChange?: (newValue: number) => void;
+  /** if readonly then only the formatValue function is used. */
+  readonly?: boolean;
+}
+
+/** Generic Input component that requires formatting and parsing functions to be passed in as props.
+ * @beta
+ */
+export const ParsedInput = React.forwardRef<HTMLInputElement, ParsedInputProps>(
+  function ParsedInput({ initialValue, formatValue, parseString, readonly, className, style, onChange }, ref) {
+    const currentValueRef = React.useRef(initialValue);
+    const isMountedRef = React.useRef(false);

```

Comment:
`isMountedRef` flag is probably not needed in this component, because `updateQuantityValueFromString` is only called in `input` events?
```suggestion
```

Reply:
Keeping in for now will do further investigation when type editor is constructed to use this component.